superset/daos/dataset.py

```diff
@@ -318,28 +318,26 @@ def create_column(
         return DatasetColumnDAO.create(properties, commit=commit)
 
     @classmethod
-    def delete_column(cls, model: TableColumn, commit: bool = True) -> TableColumn:
+    def delete_column(cls, model: TableColumn, commit: bool = True) -> None:
         """
         Deletes a Dataset column
         """
-        return cls.delete(model, commit=commit)
+        return DatasetColumnDAO.delete(model, commit=commit)
```

Comment:
This is more accurate, i.e., the `DatasetColumnDAO` rather than the `DatasetDAO` should be deleting the column.